root/src/main/java/com/example/servingwebcontent/ConnectDatabase.java

```diff
@@ -15,31 +15,66 @@ public ConnectDatabase (String database, String user, String pass) {
 		}
 	}
 
-	public ResultSet queryDB (String query) {
+	public ResultSet getMovie (String id) {
 		ResultSet rs = null;
+		String query = "select * from movies where id=?";
 		try {
-			Statement stmt = con.createStatement();
-			rs = stmt.executeQuery(query);
+			PreparedStatement stmt = con.prepareStatement(query);
+			stmt.setString(1, id);
+			rs = stmt.executeQuery();
 		} catch (SQLException e) { 
-			throw new RuntimeException("failed to query db" ,e); 
+			throw new RuntimeException("failed to query db", e); 
+		}
+
+		return rs;
+	}
+
+	public ResultSet updateMovie (String id, String column, String value) {
+		ResultSet rs = null;
+		String query = "";
+		if (column.equals("type")) {
+			query = "update movies set type = ? where id = ?";
+		}
+		else if (column.equals("name")) {
+			query = "update movies set name = ? where id = ?";
+		}
+		else if (column.equals("releasedate")) {
+			query = "update movies set releasedate = ? where id = ?";
+		}
+		else if (column.equals("duration")) {
+			query = "update movies set duration = ? where id = ?";
+		}
+		else if (column.equals("filmrating")) {
+			query = "update movies set filmrating = ? where id = ?";
+		}
+		else {
+			return rs;
+		}
+
+		try {
+			PreparedStatement preparedStmt = con.prepareStatement(query);
+			preparedStmt.setString (1, value);
+			preparedStmt.setString (2, id);
+			preparedStmt.execute();
+		} catch (SQLException e) {
+			throw new RuntimeException("failed to update movie entry", e);
 		}
 
 		return rs;
 	}
 
 	public void movieInsert (Movie newMovie) {
-		String query = "insert into movies (movieid, type, name, releasedate, duration, filmrating) values (?, ?, ?, ?, ?, ?)";
+		String query = "insert into movies (type, name, releasedate, duration, filmrating) values (?, ?, ?, ?, ?)";
 		try {
 			PreparedStatement preparedStmt = con.prepareStatement(query);
-			preparedStmt.setString (1, newMovie.getId());
-			preparedStmt.setString (2, newMovie.getType());
-			preparedStmt.setString (3, newMovie.getName());
-			preparedStmt.setString (4, newMovie.getReleaseDate());
-			preparedStmt.setInt (5, newMovie.getDuration());
-			preparedStmt.setString (6, newMovie.getFilmRating());
+			preparedStmt.setString (1, newMovie.getType());
```

Comment:
so when you had to delete the id, you had to renumber everything. You can use a variable here instead so you don't have to do that.

int i = 1;
preparedStmt.setString(i++, ...);